### PR TITLE
LPS-164617 | Add Autom. test FDSCustomView#CanBeEdited

### PR DESCRIPTION
**Pull request by mateussandes23**
Background
Create automation test for Custom Views

Test Plan

- Create a new view
- Make changes on new view
- Save changes
- Assert the changes made in the new view and the assert if the default view still the same

JIRA Ticket:
https://issues.liferay.com/browse/LPS-164617

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro [hunk 1]
```diff
@@ -7,16 +7,96 @@ definition {
 
 		Click(
 			key_column = "${key_column}",
-			locator1 = "FrontendDataSet#SELECT_FIELD");
+			locator1 = "FrontendDataSet#OPEN_SELECT_FIELD");
 
 		Click(locator1 = "FrontendDataSet#SELECT_CUSTOM_VIEW");
 	}
 
+	macro changeCustomView {
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro:15]: Could you use the `FDSCustomView.selectView()` macro while creating this?

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro [hunk 2]
```diff
@@ -28,10 +108,44 @@ definition {
...
+			value1 = "${key_filter}");
+	}
+
+	macro viewColumn {
+		VerifyElementPresent(
+			key_field = "${key_field}",
+			locator1 = "FrontendDataSet#TABLE_ITEM_COLUMN");
+	}
+
+	macro viewNoColumn {
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro:145]: duplicate macro.

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro [hunk 3]
```diff
@@ -7,16 +7,96 @@ definition {
...
+			key_optionName = "Save View",
+			locator1 = "FrontendDataSet#SELECT_OPTION");
+	}
+
+	macro saveView {
+		Click(locator1 = "FrontendDataSet#SELECT_CUSTOM_VIEW");
+
+		Click(locator1 = "FrontendDataSet#SAVE_BUTTON");
+	}
+
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro:94]: Could you put these two macros (saveView and saveCustomView) together in one using a parameter to do both functions?

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro [hunk 4]
```diff
@@ -28,10 +108,44 @@ definition {
 			locator1 = "Button#GENERAL_BUTTON_MODAL_FOOTER");
 	}
 
+	macro selectColumn {
```
Comment (mateusralv) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro:111]: I think you could use the existing `addRemoveColumn` macro here.

**Review by mateusralv — CHANGES_REQUESTED** [review]

**Comment by mateussandes23** [comment]
ci:test:frontend:nocompile

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro [hunk 5]
```diff
@@ -10,8 +20,8 @@ definition {
 		Click(locator1 = "FrontendDataSet#SELECT_CUSTOM_VIEW");
 
 		Click(
-			key_optionName = "${key_itemName}",
-			locator1 = "FrontendDataSet#SELECT_OPTION_CUSTOM_VIEW");
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro:14]: keep `key_optionName`

our standard naming is `key_${descriptor}` when we need to pass a variable to the xpath.

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro [hunk 6]
```diff
@@ -52,27 +82,11 @@ definition {
...
+			key_itemName = "Save View",
 			locator1 = "FrontendDataSet#SELECT_OPTION_CUSTOM_VIEW");
-
-		Type(
-			locator1 = "FrontendDataSet#INPUT_CUSTOM_VIEW_NAME",
-			value1 = "${key_nameView}");
-
-		Click(
-			key_name = "Save",
-			locator1 = "Button#GENERAL_BUTTON_MODAL_FOOTER");
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro:75]: We may want to keep saveViewAs. Why was it removed?

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 7]
```diff
@@ -102,41 +145,6 @@ definition {
...
-				value1 = "Default View");
-		}
-
-		task ("And Then option to select created custom view is available") {
-			AssertTextPresent(
-				locator1 = "FrontendDataSet#SELECT_OPTION_CUSTOM_VIEW",
-				value1 = "Custom View Name");
-		}
-	}
-
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:139]: Why is this test deleted?

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase [hunk 8]
```diff
@@ -75,8 +109,17 @@ definition {
 				locator1 = "FrontendDataSet#SELECT_OPTION_CUSTOM_VIEW");
 		}
 
-		task ("And When Save As the custom view") {
-			FDSCustomView.createNewView(key_nameView = "TEST");
```
Comment (john-co) [inline modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/tests/FDSCustomView.testcase:79]: My feeling is that this modification is unneeded. The scope of the task is to automate the test CanBeEdited. So any other modifications to other tests are suspicious to me.

**Review by john-co — COMMENTED** [review]
Please review comments.

File: modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro (hunk 6 above)
Comment (mateussandes23) [reply modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FDSCustomView.macro:75]: There was two macro doing the same function.

**Review by mateusralv — APPROVED** [review]

**Comment by mateusralv** [comment]
Looks good to me, @mateussandes23 could you send this PR to liferay-frontend? Thank you :)

**Comment by mateussandes23** [comment]
ci:test:sf

**Comment by mateussandes23** [comment]
ci:test:frontend:nocompile

**Comment by john-co** [comment]
Closing since PR has been submitted in https://github.com/liferay-frontend/liferay-portal/pull/2824

**Pull request closed without merging**